sdk/core/core-aborter/src/AbortSignal.ts

```diff
@@ -0,0 +1,133 @@
+import { AbortSignalLike } from "@azure/ms-rest-js";
```

Comment:
Would it make sense to inverse this relationship in the future? `@azure/ms-rest-js` seems like a big dependency to download for a single interface.

If I'm writing a function that I want to accept an AbortSignal from this package or the web native AbortSignal, I'd likely also pull in `@azure/ms-rest-js` so that I can be sure my `AbortSignalLike` matches yours, but it'd be nice if this package exported the interface.

Reply:
Also, shouldnt we import this from `@azure/core-http` instead of `@azure/ms-rest-js`?

Reply:
@chradek I agree with your intuition, the relationship should be reversed and this package should own the AbortSignalLike interface.